### PR TITLE
Make CommandExecutor pluggable

### LINKED ISSUE #901: [feature-request] Use Custom CommandExecutors
Some users request the option for using their custom CommandExecutor to execute command operator e.g. `py>`, `rb>`. They want to execute the operators on different place from Digdag where are running. Currently we don't have any extension point to load custom CommandExecutor and specify it for command operators.

### PR DESCRIPTION
**Pull request by szyn**
Follow-up #1315 and make CommandExecutor pluggable using the same approach as #1325. Close #901
We're going to add the system property `agent.command_executor.type`, which allows that users can specify command executor type. 

See also: #1315 (comment)

### CODE REVIEW

File: digdag-tests/src/test/resources/META-INF/services/io.digdag.spi.Plugin [hunk 1]
```diff
@@ -1 +1,2 @@
 acceptance.TestingAuthenticatorPlugin
+acceptance.TestingCommandExecutorPlugin
```
Comment (yoyama) [inline digdag-tests/src/test/resources/META-INF/services/io.digdag.spi.Plugin:2]: Adding new line will be better.

Comment (szyn) [reply digdag-tests/src/test/resources/META-INF/services/io.digdag.spi.Plugin]: Thank you, I added a new line. 8042e55

**Review by yoyama — APPROVED** [review]
👍

**Comment by szyn** [comment]
Thank you!

**Pull request merged**